### PR TITLE
Don't skip WAGED rebalancer calculation even the resource list is empty.

### LINKED ISSUE #1257: On resource deletion, WAGED rebalancer does not fully cleanup the in-memory tracking state.
**Describe the bug**
When a resource is removed, the WAGED rebalancer will clean up the persisted assignment state. However, it does not remove the cached IS/ResourceConfig in the Resource Change Detector. If the resource deleting is the only event happens in the cluster, then the change detector will never be refreshed. As a result, if the resource is re-created with the same name, the change detector will keep using the previous cached state to compare the configuration delta.

In the worst scenario, the new config matches the older one and the change detector thinks there is nothing changed in the cluster. As a result, the baseline won't be calculated. And the resource won't be assigned.

**To Reproduce**
(TO be confirmed, the following steps are derived from logs. Will need to add test cases to prove)
1. Create Resource A.
2. After rebalance, remove resource A.
3. Re-create resource A with the same configuration.
4. Resource A won't be assigned since the baseline calculation won't happen if no other changes happen in this cluster.

**Expected behavior**
The recreated resource A is treated as a new resource and being assigned automatically.

**Additional context**
Impacted Version: 1.0.1

This bug only impacts WAGED managed resources.

As a workaround, manually trigger rebalance (for example, touching the IdealState by changing the NUM_PARTITIONS and then changing it back), or reset the helix controller to trigger a refresh baseline rebalance.

### PR DESCRIPTION
**Pull request by jiajunwang**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

Fixes #1257 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

This change is to allow the rebalancer logic cleaning up the internal assignment state and cache if all the resources are removed from the cluster.

### Tests

- [X] The following tests are written for this issue:

TestWagedRebalance. testRecreateSameResource

- [X] The following is the result of the "mvn test" command on the appropriate module:

[ERROR] Failures:
[ERROR]   TestCustomizedViewAggregation.testCustomizedViewAggregation:413->validateAggregationSnapshot:257 expected:<true> but was:<false>
[ERROR]   TestTaskRebalancer.timeouts:204 expected:<true> but was:<false>
[INFO]
[ERROR] Tests run: 1173, Failures: 2, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:19 h
[INFO] Finished at: 2020-08-14T13:55:22-07:00
[INFO] ------------------------------------------------------------------------

Rerun

[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 30.863 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 36.773 s
[INFO] Finished at: 2020-08-14T14:20:52-07:00
[INFO] ------------------------------------------------------------------------

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/test/java/org/apache/helix/integration/rebalancer/WagedRebalancer/TestWagedRebalance.java [hunk 1]
```diff
@@ -653,6 +674,36 @@ public void testRebalancerReset() throws Exception {
     Assert.assertFalse(newEV.equals(oldEV));
   }
 
+  @Test(dependsOnMethods = "test")
+  public void testRecreateSameResource() throws InterruptedException {
```
Comment (NealSun96) [inline helix-core/src/test/java/org/apache/helix/integration/rebalancer/WagedRebalancer/TestWagedRebalance.java:678]: This is not related to the code change, right?

Comment (jiajunwang) [reply helix-core/src/test/java/org/apache/helix/integration/rebalancer/WagedRebalancer/TestWagedRebalance.java:678]: It is the issue that we want to fix.

Comment (jiajunwang) [reply helix-core/src/test/java/org/apache/helix/integration/rebalancer/WagedRebalancer/TestWagedRebalance.java:678]: Without the change, this test will fail.

**Comment by NealSun96** [comment]
Please update the PR title before merging. :) @jiajunwang

**Comment by jiajunwang** [comment]
The title is the final one which describes the change. For the other information, it has been described in the issue.

This PR has been approved and ready to merge.

**Pull request merged**